### DISCUSSION TITLE
mfront-fenics does not work after ubuntu update

### DISCUSSION THREAD

**Discussion by marcomagriluca** [thelfer/MFrontGenericInterfaceSupport#118] [General]
Good morning,

I've been using mfront/fenics for a couple of months and everything was working fine (great tool by the way!). Unfortunately, after installing some ubuntu updates (fenics was installed from ubuntu repository), mfront/fenics does not work anymore. For instance, while running the demo detailed in https://thelfer.github.io/mgis/web/mgis_fenics_nonlinear_heat_transfer.html I got the following error:

Calling FFC just-in-time (JIT) compiler, this may take some time.
['TemperatureGradient']
['HeatFlux']
['dHeatFlux_dTemperatureGradient', 'dHeatFlux_dTemperature']
Calling FFC just-in-time (JIT) compiler, this may take some time.
Calling FFC just-in-time (JIT) compiler, this may take some time.
Traceback (most recent call last):
  File "/home/marco/Desktop/Test/test.py", line 260, in <module>
    problem.register_gradient("TemperatureGradient", grad(T))
  File "/home/marco/lib/python3.10/site-packages/mgis/fenics/nonlinear_problem.py", line 227, in register_gradient
    Gradient(
  File "/home/marco/lib/python3.10/site-packages/mgis/fenics/gradient_flux.py", line 118, in __init__
    shape = ufl.shape(self.expression)
  File "/usr/lib/python3/dist-packages/ufl/operators.py", line 52, in shape
    f = as_ufl(f)
  File "/usr/lib/python3/dist-packages/ufl/constantvalue.py", line 438, in as_ufl
    raise UFLValueError("Invalid type conversion: %s can not be converted"
ufl.log.UFLValueError: Invalid type conversion: grad(Temperature) can not be converted to any UFL type.

Is it possible to bypass this in some ways, please?

Thanks a lot for help

Marco

**Comment by thelfer** [comment]
@marcomagriluca I have already encountered this and notified @bleyerj. Hope the fix will come soon.
